apps/fortuna/src/keeper.rs

```diff
@@ -297,41 +270,64 @@ pub async fn process_block_range(
         if to_block > last_block {
             to_block = last_block;
         }
+
+        process_block_batch(
+            BlockRange {
+                from: current_block,
+                to:   to_block,
+            },
+            contract.clone(),
+            gas_limit,
+            chain_state.clone(),
+        )
+        .in_current_span()
+        .await;
+
+        current_block = to_block + 1;
+    }
+}
+
+/// Process a batch of blocks for a chain. It will fetch events for the blocks for the provided batch
+/// and then try to process them one by one. If the process fails, it will retry indefinitely.
+#[tracing::instrument(name="batch", skip_all, fields(batch_from_block=block_range.from, batch_to_block=block_range.to))]
+pub async fn process_block_batch(
```

Comment:
`process_block_batch` and `process_block_range` are very similar. Maybe rename this to `process_single_block_batch`?